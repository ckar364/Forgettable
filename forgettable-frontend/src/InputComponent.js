import {TextField} from '@mui/material';
import React from 'react';
import styles from './InputComponent.module.css';
import {useState} from 'react';
import {StyledEngineProvider} from '@mui/material';

/**
 * React input component used for text entry with the possibility of
 * various stylings that can be applied.
 *
 * Note the type parameter that must be passed in. Pass in 'primary' for primary
 * input style, 'secondary' for the secondary input style, 'multiLine' for
 * the multiline style. See UI design for more details.
 */

export default function InputComponent({inputLabel, inputType, placeholder}) {
  /**
   * Retrieve state of primary input component to be rendered in a
   * responsive manner.
   * */
  const [inputState, setInput] = useState('TO DO');

  /**
   * Types allowed for input component styling.
   */
  const primary = 'primary';
  const secondary = 'secondary';
  const multiLine = 'multiLine';

  const textColour =
    getComputedStyle(document.documentElement).getPropertyValue('--ltxt1');

  return (
    <div>
      <StyledEngineProvider injectFirst>

        {inputType === primary &&
         <TextField
           className={styles.textFieldPrimary}
           label={inputLabel}
           value={inputState}
           placeholder={placeholder}
           onChange={(event) => setInput(event.target.value)}
           sx={{
             input: {color: textColour, fontSize: '23px'},
             label: {color: textColour, fontSize: '23px'},
           }}
         />
        }

        {inputType === secondary &&
         <TextField
           className={styles.textFieldSecondary}
           label={inputLabel}
           value={inputState}
           placeholder={placeholder}
           onChange={(event) => setInput(event.target.value)}
           sx={{
             input: {color: textColour, fontSize: '18px'},
             label: {color: textColour, fontSize: '18px'},
           }}
         />
        }

        {inputType === multiLine &&
         <TextField
           className={styles.textFieldMultiLine}
           label={inputLabel}
           value={inputState}
           placeholder={placeholder}
           onChange={(event) => setInput(event.target.value)}
           multiline={true}
           minRows={5}
           sx={{
             input: {color: textColour, fontSize: '16px'},
             label: {color: textColour, fontSize: '16px'},
           }}
         />
        }

      </StyledEngineProvider>
    </div>
  );
}
