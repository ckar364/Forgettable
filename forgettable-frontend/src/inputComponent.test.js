import React from 'react';
import InputComponent from './InputComponent';
import {render, cleanup, getByTestId} from '@testing-library/react';
import testRenderer from 'react-test-renderer';
import {TextField} from '@mui/material';

afterEach(cleanup);

it('renders primary input component correctly', () => {
  const primaryInput = testRenderer.create(<InputComponent inputLabel={'TO DO'} inputType={'primary'} placeholder={'TO DO PLEASE'} />).toJSON();
  expect(primaryInput).toMatchSnapshot();
});

it('renders secondary input component correctly', () => {
  const secondaryInput = testRenderer.create(<InputComponent inputLabel={'TO DO'} inputType={'secondary'} placeholder={'TO DO PLEASE'} />).toJSON();
  expect(secondaryInput).toMatchSnapshot();
});

it('renders multiLine input component correctly', () => {
  const multiLineInput = testRenderer.create(<InputComponent inputLabel={'TO DO'} inputType={'multiLine'} placeholder={'TO DO PLEASE'} />).toJSON();
  expect(multiLineInput).toMatchSnapshot();
});
