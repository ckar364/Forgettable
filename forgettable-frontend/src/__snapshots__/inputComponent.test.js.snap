// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders primary input component correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root textFieldPrimary css-7fph8o-MuiFormControl-root-MuiTextField-root"
    data-testid="primaryInput"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
    >
      TO DO
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="TO DO PLEASE"
        required={false}
        type="text"
        value="TO DO"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-7nxzgd"
        >
          <span>
            TO DO
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`renders secondary input component correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root textFieldSecondary css-o7qk7u-MuiFormControl-root-MuiTextField-root"
    data-testid="secondaryInput"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="mui-2"
      id="mui-2-label"
    >
      TO DO
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="mui-2"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="TO DO PLEASE"
        required={false}
        type="text"
        value="TO DO"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-7nxzgd"
        >
          <span>
            TO DO
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
